beaker/src/model_processing.rs

```diff
@@ -362,5 +378,67 @@ fn save_enhanced_metadata_for_file<P: ModelProcessor>(
         }
     }
 
+    // Generate depfile if requested and stamps are available
+    if let (Some(depfile_path), Some(stamp_info)) = (&config.base().depfile, stamp_info) {
+        generate_depfile_for_image::<P>(
+            result,
+            config,
+            image_path,
+            depfile_path,
+            stamp_info,
+            output_manager,
+        )?;
+    }
+
+    Ok(())
+}
+
+/// Generate stamps for the appropriate tool type
+fn generate_stamps_for_tool<P: ModelProcessor>(
+    config: &P::Config,
+    model_path: Option<&Path>,
+) -> Result<crate::stamp_manager::StampInfo> {
+    let tool_name = config.tool_name();
+
+    // Downcast to Any first, then to the specific config type
+    let config_any = config as &dyn std::any::Any;
+
+    match tool_name {
```

Comment:
Can we get rid of this match?
It's ok if not but it would be great if we did not need to update this when adding a new model